source/tests/system/nifgen_driver_api_tests.cpp

```diff
@@ -285,6 +334,80 @@ class NiFgenDriverApiTest : public ::testing::Test {
     return response.status();
   }
 
+  void allocate_named_waveform(std::string channel_name, std::string waveform_name, int waveform_size)
+  {
+    configure_output_mode(channel_name.c_str(), fgen::OutputMode::OUTPUT_MODE_NIFGEN_VAL_OUTPUT_SCRIPT);
+    ::grpc::ClientContext context;
+    fgen::AllocateNamedWaveformRequest request;
+    request.mutable_vi()->set_id(GetSessionId());
+    request.set_channel_name(channel_name);
+    request.set_waveform_name(waveform_name);
+    request.set_waveform_size(waveform_size);
+    fgen::AllocateNamedWaveformResponse response;
+
+    ::grpc::Status status = GetStub()->AllocateNamedWaveform(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    expect_api_success(response.status());
+  }
+
+  void write_named_waveform_f64(std::string channel_name, std::string waveform_name, int waveform_size, double waveform[])
+  {
+    ::grpc::ClientContext context;
+    fgen::WriteNamedWaveformF64Request request;
+    request.mutable_vi()->set_id(GetSessionId());
+    request.set_channel_name(channel_name);
+    request.set_waveform_name(waveform_name);
+    request.mutable_data()->CopyFrom(google::protobuf::RepeatedField<double>(waveform, waveform + waveform_size));
+    fgen::WriteNamedWaveformF64Response response;
+
+    ::grpc::Status status = GetStub()->WriteNamedWaveformF64(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    expect_api_success(response.status());
+  }
+
+  ViInt32 allocate_waveform(std::string channel_name, int waveform_size)
+  {
+    configure_output_mode(channel_name.c_str(), fgen::OutputMode::OUTPUT_MODE_NIFGEN_VAL_OUTPUT_SEQ);
+    ::grpc::ClientContext context;
+    fgen::AllocateWaveformRequest request;
+    request.mutable_vi()->set_id(GetSessionId());
+    request.set_channel_name(channel_name);
+    request.set_waveform_size(waveform_size);
+    fgen::AllocateWaveformResponse response;
+
+    ::grpc::Status status = GetStub()->AllocateWaveform(&context, request, &response);
+
+    EXPECT_TRUE(status.ok());
+    expect_api_success(response.status());
+
```

Comment:
No need to add a blank line here. It doesn't affect much with respect to readability.